### PR TITLE
added go mod

### PR DESCRIPTION
**Pull request by mxplusb**
Make things a bit more portable.

Signed-off-by: Mike Lloyd <mike@reboot3times.org>

### CODE REVIEW

**Comment by mxplusb** [comment]
@janisz sorry, had to fix a pgp key issue, can you please re-review?

File: go.sum [hunk 1]
```diff
@@ -0,0 +1,13 @@
+github.com/OneOfOne/xxhash v1.2.2/go.mod h1:HSdplMjZKSmBqAxg5vPj2TmRDmfkzw+cTzAElWljhcU=
+github.com/cespare/xxhash v1.1.0 h1:a6HrQnmkObjyL+Gs60czilIUGqrzKutQD6XZog3p+ko=
+github.com/cespare/xxhash v1.1.0/go.mod h1:XrSqR1VqqWfGrhpAt58auRo0WTKS1nRRg3ghfAqPWnc=
+github.com/coocood/freecache v1.1.0 h1:ENiHOsWdj1BrrlPwblhbn4GdAsMymK3pZORJ+bJGAjA=
+github.com/coocood/freecache v1.1.0/go.mod h1:ePwxCDzOYvARfHdr1pByNct1at3CoKnsipOHwKlNbzI=
+github.com/davecgh/go-spew v1.1.0 h1:ZDRjVQ15GmhC3fiQ8ni8+OwkZQO4DARzQgrnXU1Liz8=
```
Comment (cristaloleg) [inline go.sum]: We don't have this dependency in go.mod, that's strange that it's here 🤔

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,9 @@
+module github.com/allegro/bigcache
+
+go 1.12
+
+require (
+	github.com/cespare/xxhash v1.1.0 // indirect
+	github.com/coocood/freecache v1.1.0
```
Comment (cristaloleg) [inline go.mod]: Sounds like a good task for the future: remove freecache dependency.

**Comment by cristaloleg** [comment]
BTW, should we increase major version now?

File: go.sum (hunk 1 above)
Comment (mxplusb) [reply go.sum]: Those are dependencies of our testing/benchmark dependencies.

```bash
$ go mod why github.com/OneOfOne/xxhash
# github.com/OneOfOne/xxhash
github.com/allegro/bigcache/caches_bench
github.com/coocood/freecache
github.com/cespare/xxhash
github.com/cespare/xxhash.test
github.com/OneOfOne/xxhash
```

File: go.mod (hunk 2 above)
Comment (mxplusb) [reply go.mod]: It's a dependency of our benchmarking and comparison tests, not a direct dependency of bigcache itself.

**Comment by mxplusb** [comment]
> BTW, should we increase major version now?

Yeah, might as well.

File: go.mod (hunk 2 above)
Comment (cristaloleg) [reply go.mod]: Anyway that's useless 😉

**Comment by mxplusb** [comment]
Done: https://github.com/allegro/bigcache/releases/tag/v2.0.0

**Comment by maciej** [comment]
I'm scratching my head trying to understand how is this PR /making things more portable/ as you said than what I submitted in #137?

**Comment by mxplusb** [comment]
> more portable/ as you said than what I submitted in #137?

It doesn't, sorry for not accepting yours. I was addressing a couple things at the time and I didn't think about it, there was nothing wrong with your PR. My fault! Your contributions are still welcome, though!

**Pull request merged**